diff --git a/src/file_lists.cmake b/src/file_lists.cmake
index 104a296d1..68758b844 100644
--- a/src/file_lists.cmake
+++ b/src/file_lists.cmake
@@ -312,108 +312,9 @@ set(libprotoc_srcs
   ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/cpp/parse_function_generator.cc
   ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/cpp/service.cc
   ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/cpp/tracker.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/csharp/csharp_doc_comment.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/csharp/csharp_enum.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/csharp/csharp_enum_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/csharp/csharp_field_base.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/csharp/csharp_generator.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/csharp/csharp_helpers.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/csharp/csharp_map_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/csharp/csharp_message.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/csharp/csharp_message_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/csharp/csharp_primitive_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/csharp/csharp_reflection_class.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/csharp/csharp_repeated_enum_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/csharp/csharp_repeated_message_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/csharp/csharp_repeated_primitive_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/csharp/csharp_source_generator_base.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/csharp/csharp_wrapper_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/csharp/names.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/context.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/doc_comment.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/field_common.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/file.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/full/enum.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/full/enum_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/full/extension.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/full/generator_factory.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/full/make_field_gens.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/full/map_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/full/message.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/full/message_builder.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/full/message_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/full/primitive_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/full/service.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/full/string_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/generator.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/helpers.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/internal_helpers.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/java_features.pb.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/lite/enum.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/lite/enum_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/lite/extension.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/lite/generator_factory.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/lite/make_field_gens.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/lite/map_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/lite/message.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/lite/message_builder.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/lite/message_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/lite/primitive_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/lite/string_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/message_serialization.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/name_resolver.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/names.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/java/shared_code_generator.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/kotlin/file.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/kotlin/generator.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/kotlin/message.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/objectivec/enum.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/objectivec/enum_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/objectivec/extension.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/objectivec/field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/objectivec/file.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/objectivec/generator.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/objectivec/helpers.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/objectivec/import_writer.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/objectivec/line_consumer.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/objectivec/map_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/objectivec/message.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/objectivec/message_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/objectivec/names.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/objectivec/oneof.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/objectivec/primitive_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/objectivec/tf_decode_data.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/php/names.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/php/php_generator.cc
   ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/plugin.cc
   ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/plugin.pb.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/python/generator.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/python/helpers.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/python/pyi_generator.cc
   ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/retention.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/ruby/ruby_generator.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/accessors/accessor_case.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/accessors/accessors.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/accessors/default_value.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/accessors/map.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/accessors/repeated_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/accessors/singular_cord.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/accessors/singular_message.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/accessors/singular_scalar.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/accessors/singular_string.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/accessors/unsupported_field.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/accessors/with_presence.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/context.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/crate_mapping.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/enum.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/generator.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/message.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/naming.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/oneof.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/relative_path.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/rust_field_type.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/rust_keywords.cc
-  ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/rust/upb_helpers.cc
   ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/subprocess.cc
   ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/versions.cc
   ${protobuf_SOURCE_DIR}/src/google/protobuf/compiler/zip_writer.cc
diff --git a/src/google/protobuf/compiler/BUILD.bazel b/src/google/protobuf/compiler/BUILD.bazel
index 5012ee793..80c90784b 100644
--- a/src/google/protobuf/compiler/BUILD.bazel
+++ b/src/google/protobuf/compiler/BUILD.bazel
@@ -221,14 +221,6 @@ cc_library(
         "//src/google/protobuf",
         "//src/google/protobuf:port",
         "//src/google/protobuf/compiler/cpp",
-        "//src/google/protobuf/compiler/csharp",
-        "//src/google/protobuf/compiler/java",
-        "//src/google/protobuf/compiler/kotlin",
-        "//src/google/protobuf/compiler/objectivec",
-        "//src/google/protobuf/compiler/php",
-        "//src/google/protobuf/compiler/python",
-        "//src/google/protobuf/compiler/ruby",
-        "//src/google/protobuf/compiler/rust",
         "@com_google_absl//absl/log:initialize",
     ],
 )
diff --git a/src/google/protobuf/compiler/main.cc b/src/google/protobuf/compiler/main.cc
index babede8fa..97a39424a 100644
--- a/src/google/protobuf/compiler/main.cc
+++ b/src/google/protobuf/compiler/main.cc
@@ -8,6 +8,7 @@
 #include "absl/log/initialize.h"
 #include "google/protobuf/compiler/command_line_interface.h"
 #include "google/protobuf/compiler/cpp/generator.h"
+#if 0
 #include "google/protobuf/compiler/csharp/csharp_generator.h"
 #include "google/protobuf/compiler/java/generator.h"
 #include "google/protobuf/compiler/kotlin/generator.h"
@@ -17,6 +18,7 @@
 #include "google/protobuf/compiler/python/pyi_generator.h"
 #include "google/protobuf/compiler/ruby/ruby_generator.h"
 #include "google/protobuf/compiler/rust/generator.h"
+#endif
 
 #ifdef DISABLE_PROTOC_CONFIG
 #include "google/protobuf/compiler/allowlists/allowlist.h"
@@ -53,6 +55,7 @@ int ProtobufMain(int argc, char* argv[]) {
   cpp_generator.set_runtime_include_base(GOOGLE_PROTOBUF_RUNTIME_INCLUDE_BASE);
 #endif
 
+#if 0
   // Proto2 Java
   java::JavaGenerator java_generator;
   cli.RegisterGenerator("--java_out", "--java_opt", &java_generator,
@@ -106,6 +109,7 @@ int ProtobufMain(int argc, char* argv[]) {
   rust::RustGenerator rust_generator;
   cli.RegisterGenerator("--rust_out", "--rust_opt", &rust_generator,
                         "Generate Rust sources.");
+#endif
 #ifdef DISABLE_PROTOC_CONFIG
   auto cleanup = internal::DisableAllowlistInternalOnly();
 #endif  // DISABLE_PROTOC_CONFIG
